lib/reactions/fiat_rate_update.dart

```diff
@@ -30,25 +30,45 @@ Future<void> startFiatRateUpdate(
 
       if (appStore.wallet!.type == WalletType.haven) {
         await updateHavenRate(fiatConversionStore);
-      } else {
-        fiatConversionStore.prices[appStore.wallet!.currency] =
-            await FiatConversionService.fetchPrice(
-                crypto: appStore.wallet!.currency,
-                fiat: settingsStore.fiatCurrency,
-                torOnly: settingsStore.fiatApiMode == FiatApiMode.torOnly);
+        return;
       }
 
-      Iterable<CryptoCurrency>? currencies;
-      if (appStore.wallet!.type == WalletType.ethereum) {
-        currencies =
-            ethereum!.getERC20Currencies(appStore.wallet!).where((element) => element.enabled);
+      Iterable<CryptoCurrency>? currencies = [];
+      switch (appStore.wallet!.type) {
```

Comment:
so you refactored them in a switch case, which is good, but you left the old code as is, so it's now going to call the fiat API double the times it would call it

Reply:
I think this is the result of pulling main into the branch, and I guess I missed this when resolving the merge conflicts?
or maybe it was automatically merged / there were no conflicts somehow? either way good catch